proposals/lip-0031.md

```diff
@@ -84,6 +84,10 @@ The procedure for a proof-of-inclusion is as follows:
 
 In Fig. 1 we sketch a simple proof-of-inclusion for data block `data1`. The Verifier asks the Prover for a proof for `queryHashes=[leafHash(data1)]`. The Prover sends back `proof` with `size=5`, `idxs=[0001]`and `siblingHashes=[h0, h6, h4]`.
 
+### Representation of indices
+
+On an abstract level, indices in the array `idxs` are represented as bit strings. The length of such a bit string is [determined by the layer and the height of the tree](#proof-construction). For compactness, these bit strings are converted to `uint64` before being serialized such that they are [varint encoded](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0027.md#base-128-varints). As the indices can represent arbitrary nodes of the tree and not just leave nodes, different indices could represent the same integer. For example, `idxs` could contain the two indices `idx1 = "001"` and `idx2 = "0001"` (see [Fig. 2](#proof-of-inclusion)). Both would be converted to the unsigned integer `1`. To avoid this, a `1` is prepended to the bit string prior to converting to `uint64`. This way, `idx1` is converted to `0b1001 = 9` and `idx2` to `0b10001 = 17`.
```

Comment:
```suggestion
On an abstract level, indices in the array `idxs` are represented as bit strings. The length of such a bit string is [determined by the layer and the height of the tree](#proof-construction). For compactness, these bit strings are converted to `uint64` before being serialized such that they are [varint encoded](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0027.md#base-128-varints). As the indices can represent arbitrary nodes of the tree and not just leaf nodes, different indices could represent the same integer. For example, `idxs` could contain the two indices `idx1 = "001"` and `idx2 = "0001"` (see [Fig. 2](#proof-of-inclusion)). Both would be converted to the unsigned integer `1`. To avoid this, a `1` is prepended to the bit string prior to converting to `uint64`. This way, `idx1` is converted to `0b1001 = 9` and `idx2` to `0b10001 = 17`.
```